ignite/distributed/comp_models/base.py

```diff
@@ -275,8 +277,20 @@ def _do_all_gather(self, tensor: torch.Tensor) -> torch.Tensor:
     def _do_broadcast(self, tensor: torch.Tensor, src: int) -> torch.Tensor:
         pass
 
+    def _check_barrier_fn_kwargs(self, barrier_fn: Callable, kwargs_dict: Dict[str, Any]) -> Dict[str, Any]:
+        bnd_keys = set()
+        for param in signature(barrier_fn).parameters.values():
+            if param.kind == param.POSITIONAL_OR_KEYWORD:
```

Comment:
Can't we simplify this a bit? I think we can do this inside the loop, for example in the else we delete the key directly maybe? and make a bool flag maybe and raise the warning

Reply:
Thanks for your comment, @KickItLikeShika , I will try to simplify this